lib/rspec/mocks/instance_method_stasher.rb

```diff
@@ -5,6 +5,7 @@ class InstanceMethodStasher
       def initialize(object, method)
         @object = object
         @method = method
+        object.class.__persistent__ = true if object.respond_to?(:java_class)  # https://github.com/jruby/jruby/wiki/Persistence
```

Comment:
I'd prefer to conditionally define `def initialize` depending on the platform.
Still not convinced that the warning is correctly triggered - we're not adding anything to the singleton class just yet. Sounds like it is worth bringing to the attention of JRuby team.

Reply:
@pirj Looks like my first attempt doesn't work quite right even on JRuby. ☹️ I'll see if I can fix it and then you can decide if you want to do some conditional method definition hacking for performance.

Anyway, FWIW, just getting the singleton class and storing it absolutely *does* require enabling the instance proxy cache, which is what JRuby is warning about. If it wasn't enabled at that point, what could happen is that the proxied Java object could be passed to some Java code that later returns it back to Ruby, at which point JRuby would normally construct a *new* instance proxy for it, and the new proxy wouldn't know that there's an old singleton class for an old proxy for the same Java object still floating around somewhere.

Reply:
This is fine, doubles expect certain messages, and calling `respond_to?` on it may break expectations.

I'd suggest to check using `RSpec::Support::Ruby.jruby?`.